### ISSUE TITLE
nuxt generate: root-relative dynamic links are reported as 404 because crawlFetch has no baseURL

### ISSUE THREAD

**Issue by simonmaass** [harlan-zw/nuxt-link-checker#105]
### 🐛 The bug

When nuxt-link-checker runs during prerendering, root-relative dynamic links can be incorrectly reported as 404.

For example, a page containing:
`<a href="/beleuchtung-zum-rollator">Product</a>`

produces this report error: Should not respond with status code 404 (Not Found).

The product page is valid and returns 200 for both GET and HEAD when requested with its full public URL.

In the prerender path, processRouteLinks() calls getLinkResponse() without a baseURL:

```
const response = await getLinkResponse({
  link,
  timeout: config.fetchTimeout,
  fetchRemoteUrls: config.fetchRemoteUrls,
  // ...
})

```

For a relative link, crawlFetch() then executes:

```
$fetch.raw(encodeURI(link), {
  baseURL: options.baseURL,
  method: "HEAD",
  // ...
})
```

Because options.baseURL is undefined in Node during the build, the request cannot resolve /beleuchtung-zum-rollator to an absolute URL. crawlFetch() catches every request error and returns a synthetic 404:
`
.catch(() => ({ status: 404, statusText: "Not Found", headers: {} }))
`

This makes an invalid request look like a broken link.

### 🌈 Expected behavior

Internal root-relative links should be resolved against the configured site URL during prerendering, then checked with HEAD.

### ℹ️ Additional context

Pass the site URL to getLinkResponse():
```

const response = await getLinkResponse({
  link,
+ baseURL: siteConfig.url,
  timeout: config.fetchTimeout,
  fetchRemoteUrls: config.fetchRemoteUrls,
  // ...
})

```

I applied this as a local patch against nuxt-link-checker@5.2.0. After rebuilding, the report changed from 38 false 404 product-link errors to 0 errors; only unrelated underscore warnings remained.
